.github/workflows/rust.yml

```diff
@@ -0,0 +1,196 @@
+        run: node wasm/test.js
+
+      - name: Check docs
+        run: cargo clean --doc && cargo doc --all-features --no-deps && cargo deadlinks --dir target/doc
+
+  build-msrv:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Sodium packages
+        run: |
+          sudo -E apt-add-repository -y 'ppa:fsgmhoward/shadowsocks-libev'
+          sudo apt-get update
+          sudo apt-get install -y --no-install-suggests --no-install-recommends libsodium-dev
+
+      - name: Cache cargo build
+        uses: actions/cache@v1
+        with:
+          path: target
+          key: ${{ runner.os }}-msrv-cargo-build-target
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.msrv }}
+          override: true
+          profile: minimal
+
+      - name: Run tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --all-features
+      - name: Run Sodium keypair example
+        uses: actions-rs/cargo@v1
+        with:
+          command: run
+          args: --example sodium_keypair
+      - name: Run E2E tests for key util
+        run: bash tests/e2e/key_util.sh
+
+  build-nostd:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache cargo build
+        uses: actions/cache@v1
+        with:
+          path: target
+          key: ${{ runner.os }}-nostd-cargo-build-target
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.nightly }}
+          target: thumbv7m-none-eabi
+          override: true
+          profile: minimal
+
+      - name: Build library
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --lib --no-default-features --features pure --target thumbv7m-none-eabi -Z avoid-dev-deps
+
+  document:
+    needs:
+      - build
+      - build-msrv
+      - build-nostd
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Install Sodium packages
+        run: |
+          sudo -E apt-add-repository -y 'ppa:fsgmhoward/shadowsocks-libev'
+          sudo apt-get update
+          sudo apt-get install -y --no-install-suggests --no-install-recommends libsodium-dev
+
+      - name: Cache cargo build
+        uses: actions/cache@v1
+        with:
+          path: target
+          key: ${{ runner.os }}-cargo-build-target
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          profile: minimal
+
+      - name: Build docs
+        run: cargo rustdoc --all-features -- --cfg docsrs
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@releases/v3
+        with:
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
```

Comment:
It will be necessary to add `ACCESS_TOKEN` secret to the repository settings for this to work. The secret should contain a GitHub access token with a right to publish to this repo.

Reply:
Ok. I will do this.